From bd5cf3839f902b67a355669fdf1bd3231e0cb4c1 Mon Sep 17 00:00:00 2001
From: Mohammad Rafi Shaik <mohammad.rafi.shaik@oss.qualcomm.com>
Date: Mon, 9 Jun 2025 20:52:42 +0530
Subject: [PATCH] Qualcomm: Add QCS9075-IQ-EVK HiFi config

Add UCM2 configs for the Qualcomm QCS9075-IQ-EVK Board to handle:
	- I2S Speaker Amplifier
	- I2S Mic

Closes: https://github.com/alsa-project/alsa-ucm-conf/pull/576
Signed-off-by: Mohammad Rafi Shaik <mohammad.rafi.shaik@oss.qualcomm.com>
Signed-off-by: Jaroslav Kysela <perex@perex.cz>

Upstream-Status: Backport [https://github.com/alsa-project/alsa-ucm-conf/commit/bd5cf3839f902b67a355669fdf1bd3231e0cb4c1]
---
 .../qcs9075/qcs9075-iq-evk-snd-card/HiFi.conf | 29 +++++++++++++++++++
 .../qcs9075-iq-evk-snd-card.conf              |  6 ++++
 .../qcs9075/qcs9075-iq-evk-snd-card.conf      |  6 ++++
 3 files changed, 41 insertions(+)
 create mode 100644 ucm2/Qualcomm/qcs9075/qcs9075-iq-evk-snd-card/HiFi.conf
 create mode 100644 ucm2/Qualcomm/qcs9075/qcs9075-iq-evk-snd-card/qcs9075-iq-evk-snd-card.conf
 create mode 100644 ucm2/conf.d/qcs9075/qcs9075-iq-evk-snd-card.conf

diff --git a/ucm2/Qualcomm/qcs9075/qcs9075-iq-evk-snd-card/HiFi.conf b/ucm2/Qualcomm/qcs9075/qcs9075-iq-evk-snd-card/HiFi.conf
new file mode 100644
index 0000000..6673324
--- /dev/null
+++ b/ucm2/Qualcomm/qcs9075/qcs9075-iq-evk-snd-card/HiFi.conf
@@ -0,0 +1,29 @@
+SectionVerb {
+	Value {
+		TQ "HiFi"
+	}
+	EnableSequence [
+		cset "name='PRIMARY_SDR_MI2S_RX Audio Mixer MULTIMEDIA0' 1"
+		cset "name='MULTIMEDIA1 Audio Mixer TERTIARY_SDR_MI2S_TX' 1"
+	]
+}
+
+SectionDevice."Speaker" {
+	Comment "Speaker playback"
+
+	Value {
+		PlaybackPriority 100
+		PlaybackPCM "hw:${CardId},0"
+		PlaybackMixer "default:${CardId}"
+		PlaybackMixerElem "Speakers"
+	}
+}
+
+SectionDevice."Mic" {
+	Comment "Mic"
+
+	Value {
+		CapturePriority 100
+		CapturePCM "hw:${CardId},1"
+	}
+}
diff --git a/ucm2/Qualcomm/qcs9075/qcs9075-iq-evk-snd-card/qcs9075-iq-evk-snd-card.conf b/ucm2/Qualcomm/qcs9075/qcs9075-iq-evk-snd-card/qcs9075-iq-evk-snd-card.conf
new file mode 100644
index 0000000..669ba6b
--- /dev/null
+++ b/ucm2/Qualcomm/qcs9075/qcs9075-iq-evk-snd-card/qcs9075-iq-evk-snd-card.conf
@@ -0,0 +1,6 @@
+Syntax 4
+
+SectionUseCase."HiFi" {
+	File "/Qualcomm/qcs9075/qcs9075-iq-evk-snd-card/HiFi.conf"
+	Comment "HiFi quality Music"
+}
diff --git a/ucm2/conf.d/qcs9075/qcs9075-iq-evk-snd-card.conf b/ucm2/conf.d/qcs9075/qcs9075-iq-evk-snd-card.conf
new file mode 100644
index 0000000..1e49d38
--- /dev/null
+++ b/ucm2/conf.d/qcs9075/qcs9075-iq-evk-snd-card.conf
@@ -0,0 +1,6 @@
+Syntax 4
+
+SectionUseCase."HiFi" {
+	File "/Qualcomm/qcs9075/qcs9075-iq-evk-snd-card/HiFi.conf"
+	Comment "HiFi quality Music."
+}
-- 
2.34.1

